query-engine/core/src/trace_helpers/mod.rs

```diff
@@ -0,0 +1,74 @@
+use opentelemetry::sdk::export::trace::SpanData;
+use serde_json::{json, Value};
+use std::borrow::Cow;
+use std::{collections::HashMap, time::SystemTime};
+use tracing::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+const ACCEPT_ATTRIBUTES: &[&str] = &["db.statement", "itx_id", "db.type"];
+
+pub fn spans_to_json(spans: &[SpanData]) -> String {
+    let json_spans: Vec<Value> = spans.iter().map(span_to_json).collect();
+    let span_result = json!({
+        "span": true,
+        "spans": json_spans
+    });
+
+    match serde_json::to_string(&span_result) {
+        Ok(json_string) => json_string,
+        Err(_) => "".to_string(),
+    }
+}
+
+fn span_to_json(span: &SpanData) -> Value {
+    let attributes: HashMap<String, String> =
+        span.attributes
+            .iter()
+            .fold(HashMap::default(), |mut map, (key, value)| {
+                if ACCEPT_ATTRIBUTES.contains(&key.as_str()) {
+                    map.insert(key.to_string(), value.to_string());
+                }
+
+                map
+            });
+
+    // Override the name of quaint. It will be confusing for users to see quaint instead of
+    // Prisma in the spans.
+    let name: Cow<str> = if span.name == "quaint:query" {
+        "prisma:db_query".into()
+    } else {
+        span.name.clone()
+    };
+
+    json!({
+        "span": true,
+        "trace_id": format!("{}", span.span_context.trace_id()),
+        "span_id": format!("{}",span.span_context.span_id()),
+        "parent_span_id": format!("{}",span.parent_span_id),
```

Comment:
nitpick: everytime you write `format!("{}", thing)`, you can write `thing.to_string()` instead, and it will generate way less code for the compiler to compile (format!() expands to something large).

Reply:
(`to_string()` uses `Display` so the result is exactly the same)